chinese_tutorial_rvest_httr.Rmd

```diff
@@ -0,0 +1,490 @@
+title <- page %>% html_nodes("div.summary > h3") %>% html_text()
+vote <- page %>% html_nodes("span.vote-count-post > strong") %>% html_text()
+answer <- page %>% html_nodes("div.status > strong") %>% html_text
+view <- page %>% html_nodes("div.views") %>% html_attr("title")
+df <- data.frame(title=title, vote=vote,
+                 answer=answer, view=view, stringsAsFactors = FALSE)
+```
+
+1. 数据的变形、清洗
+
+3. 数据的变形、清洗
+
+我们已经得到了原始数据，但数据的类型却不是我们想要的，我们要对这些数据作分析的话，需要令`vote`、`answer`、`view`为数值型的。
+
+```R
+str(df)
+```
+
+```R
+## 'data.frame':    50 obs. of  4 variables:
+##  $ title : chr  "How to make a great R reproducible example?" "How to sort a dataframe by column(s)?" "R Grouping functions: sapply vs. lapply vs. apply. vs. tapply vs. by vs. aggregate" "How to join (merge) data frames (inner, outer, left, right)?" ...
+##  $ vote  : chr  "1870" "926" "780" "727" ...
+##  $ answer: chr  "22" "16" "9" "11" ...
+##  $ view  : chr  "158,341 views" "856,797 views" "296,448 views" "513,210 views" ...
+```
+
+```R
+df$vote <- df$vote %>% as.numeric()
+df$answer <- df$answer %>% as.numeric()
+df$view <- df$view %>% str_replace_all(pattern = "[,a-z ]+",
+                                         replacement = "") %>% as.numeric
+str(df)
+```
+
+```R
+## 'data.frame':    50 obs. of  4 variables:
+##  $ title : chr  "How to make a great R reproducible example?" "How to sort a dataframe by column(s)?" "R Grouping functions: sapply vs. lapply vs. apply. vs. tapply vs. by vs. aggregate" "How to join (merge) data frames (inner, outer, left, right)?" ...
+##  $ vote  : num  1870 926 780 727 532 526 479 425 411 404 ...
+##  $ answer: num  22 16 9 11 7 17 12 4 6 17 ...
+##  $ view  : num  158341 856797 296448 513210 60681 ...
+```
+
+在抓取数据时，可以同时完成这些简单的数据处理。
+
+1. 存储数据
+
+```R
+save(df,file="stackoverflow[r].Rdata")
+#write.table(df,file = "stackoverflow[r].csv", sep = ",", quote = FALSE,
+#            row.names = FALSE, col.names = TRUE)
+```
+
+- 实现自动翻页功能
+
+```R
+get_qInfo <- function(i){
+        require(rvest)
+        u <- sprintf("http://stackoverflow.com/questions/tagged/r?page=%d&sort=votes&pageSize=15",i)
+        page <- u %>% read_html()
+        title <- page %>% html_nodes("div.summary > h3") %>% html_text()
+        vote <- page %>% html_nodes("span.vote-count-post > strong") %>% 
+                html_text() %>% as.numeric
+        answer <- page %>% html_nodes("div.status > strong") %>% 
+                html_text %>% as.numeric
+        view <- page %>% html_nodes("div.views") %>% html_attr("title") %>%
+                str_replace_all(pattern = "[,a-z ]+", 
+                                replacement = "") %>% as.numeric
+        df <- data.frame(title=title, vote=vote,
+                 answer=answer, view=view, stringsAsFactors = FALSE)
+        return(df)
+}
+
+total <- 3
+total_info <- lapply(1:total, get_qInfo) %>% Reduce(rbind,.)
+
+save(total_info, file="stackoverflow[r].Rdata")
+```
+
+
+
+## 2. 使用 httr 包抓取数据
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings

Reply:
This problem has been solved, thank you.